finality-aleph/src/validator_network.rs

```diff
@@ -0,0 +1,41 @@
+use aleph_primitives::AuthorityId;
+use futures::future::pending;
+
+/// Network represents an interface for opening and closing connections with other Validators,
+/// and sending direct messages between them.
+///
+/// Note on Network reliability and security: it is neither assumed that the sent messages must be
+/// always delivered, nor the established connections must be secure in any way. If the Network
+/// implementation fails to deliver a message, it may assume that the send method will be called
+/// again.
+#[async_trait::async_trait]
+pub trait Network<A, D>: Send {
```

Comment:
Not sure what they exactly should be but some constraints on A and D would be nice here

Reply:
They will probably appear in implementations/uses naturally, I don't think they are necessary in the trait itself.